swagger/specification/applications/resource-manager/Applications.Core/preview/2023-10-01-preview/openapi.json

```diff
@@ -3639,6 +3639,26 @@
         }
       }
     },
+    "EnvironmentVariables": {
+      "type": "object",
+      "description": "EnvironmentVariables describes structure enabling environment variables to be set.",
+      "properties": {
+        "secrets": {
+          "type": "object",
+          "description": "The provider secrets passed as environment variables.",
+          "additionalProperties": {
+            "$ref": "#/definitions/ProviderSecret"
```

Comment:
Should we give it a more generic name rather than *Provider*Secret? The env variables are not specific to Terraform recipes and are expected to be consumed by other kinds of recipes as well where this might be used in another context with different terminology. I think this is more of a question on the [design](https://github.com/radius-project/design-notes/blob/4a768094e5541e3b1ab1716c848a06f5eb21d09f/recipe/2024-02-terraform-providers.md) than really this implementation :) @lakshmimsft, thoughts?

Reply:
Yes, ProviderSecret is not correct. My suggestion is to rename it to RecipeSecret. open to other options.
Yes, the env variables are not specific to Terraform recipes.

Reply:
Changed it to RecipeSecret.